contents/docs/plugins/build.md

```diff
@@ -124,51 +189,60 @@ async function processEvent(event, { config, cache }) {
 
 As you can see, the function receives the event before it is ingested by PostHog, adds properties to it (or modifies them), and returns the enriched event, which will then be ingested by PostHog (after all plugins run).
 
-### Global Parameters
+### processEventBatch function 
 
-PostHog automatically injects useful parameters in the `setupPlugin` and `processEvent` functions. These are:
+`processEventBatch` works just like `processEvent`, except it takes a batch of events at once, rather than one event at a time. It also returns an array of events. This is especially useful for plugins that export data out of PostHog, so that they do not need to make an HTTP request with every incoming event. 
 
-#### config
+> **Note:** Your plugin can use `processEvent` or `processEventBatch`. Currently, if both are present, `processEventBatch` will not run. 
 
-Gives you access to the PostHog config as described in `plugin.json` and configured via the PostHog interface.
+Here's the same example from above, except now using `processEventBatch`: 
 
-#### cache
-
-A way to store values that persist across `processEvent` calls. The values are stored in [Redis](https://redis.io/), an in-memory store.
+```js
+async function processEventBatch(events, { config, cache }) {
+    const counter = await cache.get('counter', 0)
+    cache.set('counter', counter + 1)
 
-Storing values is done via `cache.set`, which takes the following parameters:
+    for (let event of events) {
+      if (event.properties) {
+          event.properties['hello'] = 'world'
+          event.properties['bar'] = config.bar
+          event.properties['$counter'] = counter
+      }
+    }
 
-```
-key: string, 
-value: unknown
+    return events
+}
 ```
 
-And retrieving them uses `cache.get`, which takes:
+As you can see, since `events` is an array of events, we iterate over it to access every individual event. 
 
-```
-key: string, 
-defaultValue: unknown
-```
+> **Note:** As of right now, batch size is not yet configurable and still receives only one event (albeit in array form).
```

Comment:
:+1:

Maybe also a line saying that you don't need to return the same number of events that you get, but you must always return an array.